lib/internal/Magento/Framework/Module/Registrar.php

```diff
@@ -32,6 +32,11 @@ public static function registerModule($moduleName, $path)
         self::$modulePaths[$moduleName] = $path;
     }
 
+    public static function unregisterModule($moduleName)
```

Comment:
> Thinking further, the addition of the unregisterModule() method does not conflict with your #1562. In fact, it complements it nicely. 

I agree.

> However, testability trumps this in my view, and since we are working with global state here, some mutability is probably required.

> It should be possible to create Integration tests without mocks that run in isolation.

I think this comes down to personal preference. I don't think you're wrong, my preference is just different. I think the real smell comes from the mutability of instances of `Registrar` (addressed in #1562) which could cause the application state to be inconsistent. If #1562 is merged then I wouldn't have any major issues with `unregisterModule()`.